### PR TITLE
Expiring objects

### PR DESCRIPTION
**Pull request by JordanP**

### CODE REVIEW

File: swift_scality_backend/server.py [hunk 1]
```diff
@@ -33,6 +33,21 @@
 POLICY_STUB = object()
 
 
+class ScalityDiskFileRouter(object):
```
Comment (NicolasT) [inline swift_scality_backend/server.py]: These changes seem to be unrelated to what you're merging here?

File: swift_scality_backend/server.py [hunk 2]
```diff
@@ -33,6 +33,21 @@
...
+    Replacement for Swift's DiskFileRouter object.
+    Always returns Scality's DiskFileManager implementation.
+
+    Copied from SwiftOnFile::obj/server.py
+    """
+    def __init__(self, *args, **kwargs):
+        self.manager_cls = swift_scality_backend.diskfile.DiskFileManager(
+            *args, **kwargs)
+
+    def __getitem__(self, policy):
```
Comment (NicolasT) [inline swift_scality_backend/server.py]: Maybe the dispatching could now be done at this level instead of lower inside our bindings. Not sure that'd be backwards compatible though?

**Comment by JordanP** [comment]
We are blue: https://37.187.159.67:5443/job/swift-functional-tests/

File: swift_scality_backend/server.py (hunk 1 above)
Comment (JordanP) [inline swift_scality_backend/server.py]: It's related.

Swift's object server has a method `async_update` which access the attribute `_diskfile_router` [1]. This attribute is set up in the `setup` method of the Object server. We completely override thre setup method in our Object Server but we inherit the `async_update` method.

So we need to setup a diskfile router in our setup method, hence this change.

[1] https://github.com/openstack/swift/blob/master/swift/obj/server.py#L254

File: swift_scality_backend/server.py (hunk 2 above)
Comment (JordanP) [inline swift_scality_backend/server.py]: I am not sure I fully understand what you mean here.

And I would like to merge this PR soon because I think it is good as is and any refactoring is not 100% related to supporting expiring object.

**Comment by NicolasT** [comment]
LGTM. FWIW, when putting links to files and lines in GitHub, you can press 'y' in your browser when in the file view, which will turn the URL into an absolute one (including the commit ID). The link you used now points to a file and line in 'master', which will change over time.

**Comment by NicolasT** [comment]
Could you create a follow-up ticket for the policy selection rework?
